code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -327,6 +323,9 @@
 
 
 /mob/living/carbon/soundbang_act(intensity = 1, stun_pwr = 20, damage_pwr = 5, deafen_pwr = 15)
+	var/list/reflist = list(intensity) // Need to wrap this in a list so we can pass a reference
+	SendSignal(COMSIG_CARBON_SOUNDBANG, reflist)
+	intensity = reflist[1]

```

Comment:
not bad actually, better than anything else I can think of like a double signal chain

Reply:
perhaps we can somehow make this pattern part of the API for signals

Reply:
I shamelessly cribbed this from ninjanomnom's musings about signals a day or two ago